modules/fedimint-meta-tests/src/bin/meta-sanity-test.rs

```diff
@@ -0,0 +1,185 @@
+                client,
+                "--our-id",
+                &peer_id.to_string(),
+                "--password",
+                "notset",
+                "module",
+                "meta",
+                "get-submissions"
+            )
+            .out_json()
+            .await
+        }
+
+        async fn submit(
+            client: &Client,
+            peer_id: PeerId,
+            value: &serde_json::Value,
+        ) -> anyhow::Result<serde_json::Value> {
+            info!(%peer_id, ?value, "Peer submitting value");
+
+            cmd!(
+                client,
+                "--our-id",
+                &peer_id.to_string(),
+                "--password",
+                "notset",
+                "module",
+                "meta",
+                "submit",
+                &value.to_string(),
+            )
+            .out_json()
+            .await
+        }
+
+        pub async fn poll_value<Fut>(
+            name: &str,
+            f: impl Fn() -> Fut,
+            expected_value: serde_json::Value,
+        ) -> Result<serde_json::Value>
+        where
+            Fut: Future<Output = Result<serde_json::Value, anyhow::Error>>,
+        {
+            poll_simple(name, || async {
+                let value = f().await?;
+                if value == expected_value {
+                    Ok(value)
+                } else {
+                    bail!("Incorrect value: {}, expected: {}", value, expected_value);
+                }
+            })
+            .await
+        }
+
+        // check starting conditions
+        assert_eq!(get_consensus(&client).await?, serde_json::Value::Null);
+        assert_eq!(
+            get_submissions(&client, PeerId::from(1)).await?,
+            json! {
+                {}
+            }
+        );
+
+        let submission_value = json! {
+            { "foo": "bar" }
+        };
+        let minority_submission_value = json! {
+            { "bar": "baz" }
+        };
+
+        // check submissions visible
+        submit(&client, PeerId::from(1), &submission_value).await?;
+
+        info!("Checking submission");
+        poll_value(
+            "submission visible on same peer",
+            || async { get_submissions(&client, PeerId::from(1)).await },
+            json! {
+                {  "1": submission_value }
+            },
+        )
+        .await?;
+        poll_value(
+            "submission visible on a different peer",
+            || async { get_submissions(&client, PeerId::from(3)).await },
+            json! {
+                {  "1": submission_value }
+            },
+        )
+        .await?;
+
+        // form a consensus with a minority vote
+        submit(&client, PeerId::from(0), &minority_submission_value).await?;
+        submit(&client, PeerId::from(2), &submission_value).await?;
+        assert_eq!(
+            submit(&client, PeerId::from(3), &submission_value).await?,
+            serde_json::Value::Bool(true)
+        );
```

Comment:
nit: Does this `true` carry any meaning? Seems to be statically returned.

https://github.com/fedimint/fedimint/blob/fedd08cb69cad763fb3511e15f08e252af5a1ec0/modules/fedimint-meta-client/src/cli.rs#L99-L113

Should we test it for all submissions or maybe none?